src/spartan/polys/power.rs

```diff
@@ -30,6 +30,24 @@ impl<Scalar: PrimeField> PowPolynomial<Scalar> {
     }
   }
 
+  /// Create powers the following powers of `t`:
+  /// [t^{2^0}, t^{2^1}, ..., t^{2^{ell-1}}]
+  pub(crate) fn powers(t: &Scalar, ell: usize) -> Vec<Scalar> {
+    let mut t_pow = vec![Scalar::ONE; ell];
+    t_pow[0] = *t;
+    for i in 1..ell {
+      t_pow[i] = t_pow[i - 1].square();
+    }
+    t_pow
+  }
+
+  /// Creates the evals corresponding to a `PowPolynomial` from an already-existing vector of powers.
+  /// `t_pow.len() > ell` must be true.
+  pub(crate) fn evals_with_powers(powers: &[Scalar], ell: usize) -> Vec<Scalar> {
+    let t_pow = powers[..ell].to_vec();
```

Comment:
This is a clone: not sure if this is usable here, but https://docs.rs/ref-cast/latest/ref_cast/ allows (safe-)casting a `&[Scalar]` to a `&EqPolynomial` if it is enough to get us by.

Reply:
It doesn't quite work in practice. I did some other juggling to remove the remaining clone, I think, in the latest.